addons/product/models/product_pricelist_item.py

```diff
@@ -372,10 +373,10 @@ def _compute_price(self, product, quantity, uom, date):
         if self.compute_price == 'fixed':
             price = convert(self.fixed_price)
         elif self.compute_price == 'percentage':
-            base_price = self._compute_base_price(product, quantity, uom=uom, date=date)
+            base_price = self._compute_base_price(product, quantity, currency, uom=uom, date=date)
             price = (base_price - (base_price * (self.percent_price / 100))) or 0.0
         else:
-            base_price = self._compute_base_price(product, quantity, uom=uom, date=date)
+            base_price = self._compute_base_price(product, quantity, currency, uom=uom, date=date)
```

Comment:
IMHO, `currency` should only be propagated up to `_compute_price`, not to `_compute_base_price` :thinking:

Reply:
In fact, not even in `_compute_price`. This is only ever called when there is a `pricelist_rule` =]